### DISCUSSION TITLE
Open duration of contact sensor

### DISCUSSION THREAD

**Discussion by sagarpandit68** [homebridge-simplisafe3/homebridge-simplisafe3#220] [Q&A] [answered]
Hello,
   I am new to Homekit but I want to create an automation where if SimpliSafe contact sensor is open for certain duration then switch off ecobee. Is there any way I can access the duration of open information? I will be happy to hear, if anyone has any other suggestion on achieving this. Thanks in advance and best wishes

**Comment by shamoon** [comment, marked as answer]
Hi @sagarpandit68 yea this is often cited as a shortfall of HomeKit, it’s not supported natively AFAIK. There may be other solutions but what I (and many others) do is use another Homebridge plug-in called [Homebridge Dummy](https://github.com/nfarina/homebridge-dummy) which lets you setup virtual switches with a timer.

There are different ways to use this but for me for example, with a switch that has a 10 second timer:
Automation 1: when the [simplisafe] contact sensor opens —> turn on light & turn virtual switch on
Automation 2: when the virtual switch turns off—> turn off the light

The only other point worth noting is that, as mentioned in the readme, contact sensors don’t push their state via SS API so instead we have to poll for them every X seconds (`sensorRefresh` time), so there is often a delay in real world circumstances.

Hope that helps

**Comment by sagarpandit68** [comment]
Thanks a lot for this great response. I will try this. Best wishes

Sagar
